test/es6module/exportBinding.js

```diff
@@ -0,0 +1,57 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+export class A1 {
+    attemptBindingChange() { A1 = 1; }
+    bindingModified() { return A1 !== 1; }
+}
+var A1Inst = new A1();
+
+assert.throws(function () { A1Inst.attemptBindingChange(); }, TypeError, 'Exported class name captured inside class is const', "Assignment to const");
+assert.doesNotThrow(function () { A1 = 1; }, 'Exported class name decl binding is mutable outside class');
+assert.throws(function () { A1Inst.attemptBindingChange(); }, TypeError, 'Mutation of class decl did not change constness', "Assignment to const");
+assert.areEqual(true, A1Inst.bindingModified(), 'Captured class name observes mutation');
+assert.throws(function () { A1Inst.attemptBindingChange(); }, TypeError, 'Exported class name captured inside class is still const', "Assignment to const");
+
+class A2 {
+    attemptBindingChange() { A2 = 1; }
+    bindingModified() { return A2 !== 1; }
+}
+var A2Inst = new A2();
+export { A2 };
+
+assert.throws(function () { A2Inst.attemptBindingChange(); }, TypeError, 'Delay exported class name captured in class is const', "Assignment to const");
+assert.doesNotThrow(function () { A2 = 1; }, 'Delay exported class name decl binding is mutable outside class');
+assert.throws(function () { A2Inst.attemptBindingChange(); }, TypeError, 'Mutation of class decl did not change constness', "Assignment to const");
+assert.areEqual(true, A2Inst.bindingModified(), 'Captured class name observes mutation');
+assert.throws(function () { A2Inst.attemptBindingChange(); }, TypeError, 'Delay exported class name captured in class is still const', "Assignment to const");
+
+export let B1 = class B1 {

```

Comment:
I'd like one more set of these if you don't mind where the let binding name is different from the class expression name. `export let C1 = class NotC1 {` for example.

Reply:
Done. Also fixed my conditional language which was inverted :)